helpers.go

```diff
@@ -26,6 +26,16 @@ type Optional interface {
 	IsDefined() bool
 }
 
+// UnknownsUnmarshaler provides a method to unmarshal unknown struct fileds and save them as you want
+type UnknownsUnmarshaler interface {
+	UnmarshalUnknown(in *jlexer.Lexer, key string)
+}
+
+// UnknownsMarshaler provides a method to write additional struct fields
+type UnknownsMarshaler interface {
+	MarshalUnknowns(w *jwriter.Writer, first bool)
+}
+
```

Comment:
Why it's `UnmarshalUnknown` (singular) but `MarshalUnknowns` (plural)?

IMHO, the names should be more JSON- or even EasyJSON-specific, as they're expected to be implemented in the user's code.

Reply:
> Why it's UnmarshalUnknown (singular) but MarshalUnknowns (plural)?

the first one unmarshals one field when the second marshals multiple fields.

 > more JSON- or even EasyJSON-specific

it makes sense. what do you propose? UnmarshalUnknownEasyJSON + MarshalUnknowns EasyJSON ?

i think we can describe your proposal in another [MR](https://github.com/mailru/easyjson/pull/269) because this one is already merged.